File: mobile/src/constants/categories.ts
/**
 * Category definitions for news and podcasts
 * Matches backend category mapping
 */

import { ComponentProps } from 'react';
import { Ionicons } from '@expo/vector-icons';

// Type-safe Ionicons name
export type IconName = ComponentProps<typeof Ionicons>['name'];

export interface Category {
  id: string;          // English API name
  name: string;        // Korean display name
  icon: IconName;      // Icon name from Ionicons (type-safe)
  color: string;       // Accent color for this category
}

export const CATEGORIES: Category[] = [
  {
    id: 'politics',
    name: '정치',
    icon: 'business',
    color: '#EF4444', // Red
  },
  {
    id: 'economy',
    name: '경제',
    icon: 'trending-up',
    color: '#10B981', // Green
  },
  {
    id: 'society',
    name: '사회',
    icon: 'people',
    color: '#8B5CF6', // Purple
  },
  {
    id: 'culture',
    name: '문화',
    icon: 'color-palette',
    color: '#EC4899', // Pink
  },
  {
    id: 'international',
    name: '국제',
    icon: 'globe',
    color: '#3B82F6', // Blue
  },
  {
    id: 'local',
    name: '지역',
    icon: 'location',
    color: '#F59E0B', // Amber
  },
  {
    id: 'sports',
    name: '스포츠',
    icon: 'trophy',
    color: '#06B6D4', // Cyan
  },
  {
    id: 'tech',
    name: 'IT/과학',
    icon: 'hardware-chip',
    color: '#6366F1', // Indigo
  },
];

export const getCategoryById = (id: string): Category | undefined => {
  return CATEGORIES.find((cat) => cat.id === id);
};

export const getCategoryByName = (name: string): Category | undefined => {
  return CATEGORIES.find((cat) => cat.name === name);
};

export const getCategoryColor = (id: string): string => {
  return getCategoryById(id)?.color || '#94A3B8';
};

export const getCategoryIcon = (id: string): IconName => {
  return (getCategoryById(id)?.icon || 'newspaper') as IconName;
};

File: flutter/lib/core/constants/categories.dart
import 'package:flutter/material.dart';

import '../theme/app_icons.dart';

class Category {
  final String id;        // api_name (e.g. 'economy')
  final String ko;        // 한글 표시명
  final Color color;      // 카테고리 전용 색 (muted editorial)
  final AppIconName icon;

  const Category({
    required this.id,
    required this.ko,
    required this.color,
    required this.icon,
  });
}

// id 는 backend CATEGORY_MAP 의 api_name 과 일치해야 함 (Supabase news_cards.category 컬럼).
// 지역·스포츠는 backend 가 수집하지 않으므로 제외.
const kCategories = <Category>[
  Category(id: 'politics',      ko: '정치',    color: Color(0xFFB54A3A), icon: AppIconName.building),
  Category(id: 'economy',       ko: '경제',    color: Color(0xFF4A7C59), icon: AppIconName.trending),
  Category(id: 'society',       ko: '사회',    color: Color(0xFF6B4E8A), icon: AppIconName.people),
  Category(id: 'culture',       ko: '문화',    color: Color(0xFFB85A75), icon: AppIconName.palette),
  Category(id: 'international', ko: '국제',    color: Color(0xFF3D6B8C), icon: AppIconName.globe),
  Category(id: 'tech',          ko: 'IT/과학', color: Color(0xFF5A5FB0), icon: AppIconName.chip),
];

final Map<String, Category> kCategoryById = {
  for (final c in kCategories) c.id: c,
};
